evalml/tests/automl_tests/test_automl.py

```diff
@@ -5102,6 +5102,25 @@ def test_pipeline_parameter_warnings_only_parameternotused(
     assert str(warning) == "Big bad warning"
 
 
+def test_pipeline_parameter_warnings_with_other_types():
+    # Data that will trigger a UserWarning from `TargetLeakageDataCheck`
+    X = pd.DataFrame({"a": [1, 2, 2, 1, 2, 1, 2, 2, 1, 1] * 505})
+    y = [0.946, 0.972, 1.154, 0.954, 0.969, 1.222, 1.038, 0.999, 0.973, 0.897] * 505
+
+    with pytest.warns(None) as w:
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="regression",
+            allowed_model_families=["random_forest", "linear_model"],
```

Comment:
We're using random forest and linear model families, so I think the params you passed for Decision Tree won't be used.

I think we can shorten this to just set `max_iterations=2` rather than defining `allowed_model_families`, `max_batches`, and `pipeline_parmaeters`

Reply:
The params for Decision Tree is part of the test, making sure that `ParameterNotUsed` warnings and other kinds of warnings that come up on the same run will both be thrown, since that was the original issue.

Reply:
Wow that completely slipped my mind. Thanks for clarifying!